src/esp/gfx/Shader.h

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <Magnum/GL/AbstractShaderProgram.h>
+
+#include "esp/core/esp.h"
+
+namespace esp {
+namespace gfx {
+
+class Drawable;
+class DrawableGroup;
+class RenderCamera;
+
+// TODO: string type to allow for dynamically added shader types
```

Comment:
Great idea, we can definitely use that!

I wanted to clarify, these strings won't actually be used in the render loop, instead only used on new shader creation or setting a new config for a created `Shader`. `ShaderType` only exists so that we know which factory function to call when given a `ShaderConfig`. The actual shader lookup based on shader UUID will be done in `ShaderManager` (see #454).